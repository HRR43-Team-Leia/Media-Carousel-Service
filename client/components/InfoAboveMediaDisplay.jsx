import React, { useState, useEffect } from 'react';
import styled from 'styled-components';

const Wrapper = styled.section`
padding: 4em;
background: black;
display: flex;
`;

const experienceHardCodedData = {
  city: 'Mexico City',
  state: null,
  country: 'Mexico',
  category: 'Arts and Culture',
  activity: 'History'
}

function InfoAboveMediaDisplay () {
  const [experienceId, setExperienceId] = useState(experienceHardCodedData);

  // useEffect to perform API call to get data
  // useEffect(() => {

  // });

  return (
    <div>
      <Wrapper>
        <ul className="listCityAndActivity">
          <li className="listItem">
            <a>
              <span className="city-country">Mexico City, Mexico</span>
            </a>
          </li>
          <li className="listItem">
            <span className="slash"> / </span>
          </li>
          <li className="listItem">
            <a>
              <span className="category">Arts and Culture</span>
            </a>
          </li>
          <li className="listItem">
            <span className="slash"> / </span>
          </li>
          <li className="listItem">
            <a>
              <span className="activity">Art Walk</span>
            </a>
          </li>
        </ul>
      </Wrapper>
    </div>
  )

}


export default InfoAboveMediaDisplay;

