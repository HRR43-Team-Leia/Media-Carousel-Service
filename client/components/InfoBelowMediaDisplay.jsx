import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  faStar, faClock, faUserFriends, faReceipt, faComments,
} from '@fortawesome/free-solid-svg-icons';

const Wrapper = styled.section`
box-sizing: border-box;
width: 100vw;
background: black;
display: flex;
padding-top: 30px;
padding-bottom: 40px;
`;

const FirstCol = styled.div`
  flex-basis: 35%;
  flex-wrap: wrap;
  justify-content: flex-start;
`;

const SecondCol = styled.div`
  display: flex;
  flex-wrap: wrap;
  flex-basis: 65%;
`;

const IndvDetail = styled.div`
display: flex;
flex-direction: column;
flex-basis: 25%;
padding-right: 35px;
`;

const StarsAndReviews = styled.span`
padding-right: 8px;
`;

const DetailsGroup = styled.div`
display: flex;
flex-direction: row;
justify-content: flex-around;
padding: 15px 0 0 0;
`;

class InfoBelowMediaDisplay extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      navbarBottomPosition: null,
    };

    this.listenScrollEvent = this.listenScrollEvent.bind(this);
  }

  componentDidMount() {
    window.addEventListener('scroll', this.listenScrollEvent);
  }

  listenScrollEvent() {
    const navBar = document.querySelector('.infoBelowBottom').getBoundingClientRect();
    this.setState({ navbarBottomPosition: navBar.bottom });
    const { updateNavBarPosition } = this.props;
    const { navbarBottomPosition } = this.state;
    updateNavBarPosition(navbarBottomPosition);
  }

  render() {
    const { languages } = this.props;
    const { includes } = this.props;

    return (
      <Wrapper className="infoBelowBottom">
        <section className="infoBelow">
          <FirstCol>
            <div className="activity-classification">{this.props.experience.activity}</div>
            <div className="activity-title">{this.props.experience.title}</div>
            <div className="activity-city-state">
              {this.props.experience.city}
,
              {' '}
              {this.props.experience.country}
            </div>
            <div className="average-rating">
              <StarsAndReviews>
                {this.props.experience.averageRating}
              </StarsAndReviews>
              <StarsAndReviews>
                <FontAwesomeIcon icon={faStar} size="sm" />
              </StarsAndReviews>
              (
              {this.props.experience.numberOfReviews}
                )
              <StarsAndReviews />
            </div>
            <div>{this.props.experience.tags}</div>
          </FirstCol>
          <SecondCol>
            <DetailsGroup>
              <IndvDetail>
                <div className="duration-icon">
                  <FontAwesomeIcon icon={faClock} size="xs" />
                </div>
                <div className="duration-title-time">
                  <div className="duration-title" style={{ color: '#DCDCDC' }}>Duration</div>
                  <div className="hours">
                    {this.props.experience.duration}
                    {' '}
hours
                  </div>
                </div>
              </IndvDetail>
              <IndvDetail>
                <div className="group-size-icon">
                  <FontAwesomeIcon icon={faUserFriends} size="sm" />
                </div>
                <div className="group-size-title-time">
                  <div className="group-size-title" style={{ color: '#DCDCDC' }}>Group size</div>
                  <div className="group-size">
Up to
                    {' '}
                    {this.props.experience.groupSize}
                    {' '}
people
                  </div>
                </div>
              </IndvDetail>
              <IndvDetail>
                <div className="includes-icon">
                  <FontAwesomeIcon icon={faReceipt} size="sm" />
                </div>
                <div className="includes-title-items">
                  <div className="includes-title" style={{ color: '#DCDCDC' }}>Includes</div>
                  <div className="includes-items">
                    {includes.map((include) => (
                      <div key={include}>
                        {include}
,
                      </div>
                    ))}

                  </div>
                </div>
              </IndvDetail>
              <IndvDetail>
                <div className="hosted-in-icon">
                  <FontAwesomeIcon icon={faComments} size="sm" />
                </div>
                <div className="hosted-in-title">
                  <div className="duration-title" style={{ color: '#DCDCDC' }}>Hosted in</div>
                  <div className="languages">
                    {languages.map((language) => (
                      <div key={language}>
                        {language}
,
                      </div>
                    ))}
                  </div>
                </div>
              </IndvDetail>
            </DetailsGroup>
          </SecondCol>
        </section>
      </Wrapper>
    );
  }
}


export default InfoBelowMediaDisplay;
