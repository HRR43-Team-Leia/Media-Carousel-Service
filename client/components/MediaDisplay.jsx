import React from 'react';
import styled from 'styled-components';


class MediaDisplay extends React.Component {
  constructor() {
    super();
  }

  render() {
    return (
      <div class="image-wrapper">
        <section class="image-section">
          <div class="image-cover">
            <img src="https://picsum.photos/id/1018/312/416" alt="an image of something" />
          </div>
          <div class="image-cover">
            <img src="https://picsum.photos/id/1047/312/416" alt="an image of something" />
          </div>
          <div class="vertical-stack-images">
            <div class="image-cover">
              <img src="https://picsum.photos/id/1055/200/300" alt="an image of something" />
            </div>
            <div class="image-cover">
              <img src="https://picsum.photos/id/106/156/208" alt="an image of something" />
            </div>
          </div>
          <div class="image-cover">
            <img src="https://picsum.photos/id/1069/312/416" alt="an image of something" />
          </div>
        </section>
      </div>
    )
  }
};

export default MediaDisplay;