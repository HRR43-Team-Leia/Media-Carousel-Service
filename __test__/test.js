import React from 'react';
import Enzyme, { render, shallow, mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter()});

import InfoBelowMediaDisplay from '../client/components/InfoBelowMediaDisplay';

import MediaDisplay from '../client/components/MediaDisplay';

import InfoAboveMediaDisplay from '../client/components/InfoAboveMediaDisplay';



describe('InfoBelowMediaDisplay should render correctly ', () => {
  it('should be defined', () => {
    expect(InfoBelowMediaDisplay).toBeDefined();
  });

  it('InfoBelowMediaDisplay should shallow render', () => {
    const Wrapper = shallow(<InfoBelowMediaDisplay />)
    // if passing props props={[]}
    expect(Wrapper).toMatchSnapshot()
  });
})

describe('MediaDisplay should render correctly ', () => {
  it('should be defined', () => {
    expect(MediaDisplay).toBeDefined();
  });

  it('MediaDisplay should shallow render', () => {
    const Wrapper = shallow(<MediaDisplay />)

    expect(Wrapper).toMatchSnapshot()
  });

  it('expect 5 images to be rendered', () => {
    const wrapper = shallow(<MediaDisplay />)

    expect(wrapper.find('img')).toHaveLength(5)
  });
});

describe('InfoAboveMediaDisplay should render correctly ', () => {
  it('should be defined', () => {
    expect(InfoAboveMediaDisplay).toBeDefined();
  });

  it('InfoAboveMediaDisplay should shallow render', () => {
    const Wrapper = shallow(<InfoAboveMediaDisplay />)

    expect(Wrapper).toMatchSnapshot()
  });
})


