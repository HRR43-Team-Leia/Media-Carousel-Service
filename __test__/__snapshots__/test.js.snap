// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoAboveMediaDisplay should render correctly  InfoAboveMediaDisplay should shallow render 1`] = `ShallowWrapper {}`;

exports[`InfoBelowMediaDisplay should render correctly  InfoBelowMediaDisplay should shallow render 1`] = `ShallowWrapper {}`;

exports[`MediaDisplay should render correctly  MediaDisplay should shallow render 1`] = `ShallowWrapper {}`;
